lab-anthony/test/drink-router-success-test.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const expect = require('chai').expect;
+const request = require('superagent');
+const server = require('../server');
+const storage = require('../lib/storage');
+const Drink = require('../model/drink');
+const port = process.env.PORT || 3000;
+const baseUrl = `localhost:${port}/api/drink`;
+
+describe('testing module drink-router', function(){
+  before((done)=>{
+    if(!server.isRunning){
+      server.listen(port, ()=>{
+        console.log('server running on port', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+  after((done)=>{
+    if (server.isRunning){
+      server.close(()=>{
+        server.isRunning = false;
+        console.log('server shutdown');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('testing POST /api/drink', function(){
+    after((done)=>{
+      storage.pool = {};
+      done();
+    });
+
+    it('should return a drink', function(done){
+      request.post(baseUrl)
+      .send({drinkName:'Macchiato', cafeName:'Cafe Javasti'})
+      .end((err, res)=>{
+        expect(res.status).to.equal(200);
+        expect(res.body.drinkName).to.equal('Macchiato');
+        expect(res.body.cafeName).to.equal('Cafe Javasti');
+        done();
+      });
+    });
+  });
+
+  describe('testing GET /api/drink', function(){
+    before((done)=>{
```

Comment:
This before/after pattern should probably be done with a beforeEach and an afterEach.